### DISCUSSION TITLE
About support remote repository

### DISCUSSION THREAD

**Discussion by Hebe970217** [clangd/clangd#1059] [Q&A]
I have a question. 
In the recent discussion on LSP's official website, there is a problem about remote code warehouse. 
As we know, the file path transmitted by LSP protocol is standard URI. 
Will Clangd support virtual protocol file system in the future. 
For the current difficulties, the most dependence comes from how Clangd retrieves the runtime dependent library files of the file system installation environment. 
Or Clangd doesn't intend to integrate the remote warehouse code protocol. 
Because I saw in the support of vscode that they used other ways to implement relevant functions (docker + VFS).

Thank you for answering questions from anyone.

**Comment by sam-mccall** [comment]
Can you link to this discussion or to this remote warehouse code protocol?
Is it part of LSP, or just an idea?

Internally clangd does support virtual file systems, but not over any LSP-like protocol.

**Reply by Hebe970217** [reply]
see https://github.com/microsoft/vscode-docs/blob/main/remote-release-notes/v1_56.md
from vscode,
and https://github.com/microsoft/language-server-protocol/issues/1264 from lsp protocol

If the current Clangd supports virtual file system, system include should not be found?
The scenario I think about is that on X86 computers, you can link Clangd on ARM computers through the network and use the operations from Clangd on ARM computer.
